libs/libGamedata/gameData/CampaignDescription.cpp

```diff
@@ -34,7 +34,16 @@ CampaignDescription::CampaignDescription(const kaguya::LuaRef& table)
     lua::validatePath(luaFolder);
     mapNames = luaData.getOrDefault("maps", std::vector<std::string>());
 
-    luaData.checkUnused();
+    kaguya::LuaRef entry = table["selectionMap"];
+    if(entry.type() == LUA_TTABLE)
+    {
+        selectionMapData = std::make_optional<SelectionMapInputData>(entry);
+    }
```

Comment:
Why the removal of `checkUnused`? What about 
```suggestion
    selectionMapData = luaData.getOrDefault("selectionMap", std::optional<SelectionMapInputData>());
    luaData.checkUnused();
```

If that doesn't work out of the box we need a `getOptional<SelectionMapInputData>` for `CheckedLuaTable` based on `getOrDefault`

Reply:
added getoptional